### PR TITLE
Add support for STRALGO

### PR DESCRIPTION
**Pull request by dengliming**
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/lettuce-io/lettuce-core/blob/master/.github/CONTRIBUTING.md).
- [x] You have created a feature request first to discuss your contribution intent. Please reference the feature request ticket number in the pull request.
- [x] You use the code formatters provided [here](https://github.com/lettuce-io/lettuce-core/blob/master/formatting.xml) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.

 
#1280

### CODE REVIEW

File: src/test/java/io/lettuce/core/commands/StringCommandIntegrationTests.java [hunk 1]
```diff
@@ -238,4 +241,39 @@ void time() {
         Long.parseLong(time.get(0));
         Long.parseLong(time.get(1));
     }
-}
+
+    @Test
+    void strAlgo() {
```
Comment (mp911de) [inline src/test/java/io/lettuce/core/commands/StringCommandIntegrationTests.java:246]: Please split this method into multiple test methods so we can test each scenario individually.

**Comment by mp911de** [comment]
Thanks for the pull request. `STRALGO` indicates a series of commands where it makes sense to model each command with its own method. So rather `LCS` being part of `StrAlgoArgs`, we should have a `stralgoLcs(StrAlgoArgs)` method and remove `LCS` from `StrAlgoArgs`. So `StringAlgorithm` isn't needed as type.

It also makes sense to provide a `Charset` option for the String-based variant.

**Comment by dengliming** [comment]
> Thanks for the pull request. `STRALGO` indicates a series of commands where it makes sense to model each command with its own method. So rather `LCS` being part of `StrAlgoArgs`, we should have a `stralgoLcs(StrAlgoArgs)` method and remove `LCS` from `StrAlgoArgs`. So `StringAlgorithm` isn't needed as type.
> 
> It also makes sense to provide a `Charset` option for the String-based variant.

Thanks for reviewing. I will fix it later.

File: src/test/java/io/lettuce/core/commands/StringCommandIntegrationTests.java (hunk 1 above)
Comment (dengliming) [reply src/test/java/io/lettuce/core/commands/StringCommandIntegrationTests.java:246]: fixed at last commit.

**Comment by dengliming** [comment]
@mp911de I'm a little confused here. `It also makes sense to provide a Charset option for the String-based variant.` You mean like this?
```
public StrAlgoArgs charset(Charset charset) {
        this.charset = charset;
        return this;
    }
```
io.lettuce.core.StrAlgoArgs#build
```
 for (String key : keys) {
            args.add(key.getBytes(charset));
        }
```

**Comment by mp911de** [comment]
Yes, something along the lines but for `STRINGS` and not `KEYS`, so the code would be `strings(Charset charset, String... strings)` while `keys(…)` remains as-is.

**Comment by dengliming** [comment]
> Yes, something along the lines but for `STRINGS` and not `KEYS`, so the code would be `strings(Charset charset, String... strings)` while `keys(…)` remains as-is.

@mp911de Thanks. fixed it at lastest commit. please re-review.

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged, polished, and backported now.

**Pull request closed without merging**